### ISSUE TITLE
ImportError: failed to find libmagic

### ISSUE THREAD

**Issue by Positron010** [thiswillbeyourgithub/wdoc#3]
Hey,
I installed the latest version using the `pip` command, but I get this error whenever I run it (name of home folder removed):

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\...\AppData\Local\Programs\Python\Python312\Scripts\wdoc.exe\__main__.py", line 4, in <module>
  File "C:\Users\...\AppData\Local\Programs\Python\Python312\Lib\site-packages\WDoc\__init__.py", line 72, in <module>
    from .WDoc import WDoc
  File "C:\Users\...\AppData\Local\Programs\Python\Python312\Lib\site-packages\WDoc\WDoc.py", line 6, in <module>
    from .utils.logger import whi, yel, red, md_printer, logger, set_USAGE_as_docstring, log_dir, cache_dir
  File "C:\Users\...\AppData\Local\Programs\Python\Python312\Lib\site-packages\WDoc\utils\__init__.py", line 1, in <module>
    from . import batch_file_loader, loaders, misc, prompts, tasks, customs
  File "C:\Users\...\AppData\Local\Programs\Python\Python312\Lib\site-packages\WDoc\utils\batch_file_loader.py", line 19, in <module>
    import magic
  File "C:\Users\...\AppData\Local\Programs\Python\Python312\Lib\site-packages\magic\__init__.py", line 209, in <module>
    libmagic = loader.load_lib()
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\...\AppData\Local\Programs\Python\Python312\Lib\site-packages\magic\loader.py", line 49, in load_lib
    raise ImportError('failed to find libmagic.  Check your installation')
ImportError: failed to find libmagic.  Check your installation

```

Thanks

**Comment by thiswillbeyourgithub** [comment]
Hi, so sorry you're giving a try at WDoc today, I've been finding issues left and right since yesterday and pushing a lot. Bad timing!

Thanks for reporting this issue. I wasn't aware magic could have such issue so I'm making it optional as we speak

**Comment by thiswillbeyourgithub** [comment]
Pushed to Pypi as 1.8.22. I'll close this but don't hesitate to tell me if it still fails or something, that's the only way for me to fix this kind of bug :)!

**Comment by Positron010** [comment]
Hey,
Thanks for the reply and your efforts. I unfortunately still get the same error after upgrading to the latest version.

**Comment by thiswillbeyourgithub** [comment]
This might be due to using Python 3.12. I've only tested it on 3.11.7.
Would you mind testing it on Python 3.11.7? Maybe with uv or uvx

**Comment by thiswillbeyourgithub** [comment]
Oh, actually I misread your trace. I just uploaded ~~1.8.26~~ 1.8.27 hot out of the oven for you

**Comment by Positron010** [comment]
Hey,
The aforementioned issue has been resolved with this upgrade, and the script can run and show the help, albeit it shows this log whenever it is runned:

```
Couldn't import optional package 'ftlangdetect', trying to import langdetect (but it's much slower): 'No module named 'ftlangdetect''
WARNING:langchain_community.utils.user_agent:USER_AGENT environment variable not set, consider setting it to identify your requests.
C:\Users\...\AppData\Local\Programs\Python\Python312\Lib\site-packages\pydantic\_internal\_fields.py:132: UserWarning: Field "model_name" in HuggingFaceInferenceAPIEmbeddings has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
  warnings.warn(

```

Thanks, and hope this helps.

**Comment by thiswillbeyourgithub** [comment]
Yeah, so basically there are two packages that are used to infer the language in a given string. And the idea is that I first parse a document, then pass it through that detector. And if it's really unsure about the language, then it probably means that the parsing was of poor quality. The first detector I used was quite slow and then I discovered the fast text one and it's faster but can be tricky to get working on some setups so I decided to try to install both and just warn the user if it's not able to use the fastest one.

After your message, I decided to only print those messages if verbose is true. Thank you.
